### DISCUSSION TITLE
How to compile arm64-linux packages under Windows platform?

### DISCUSSION THREAD

**Discussion by WangZP10969** [microsoft/vcpkg#24173] [Q&A] [answered]
I am trying to compile qt5 under Windows10 platform: arm64-linux
I have Visual Studio Community 2022 installed and I have C++ for desktop development and tools for Linux and embedded development with C++
I also have Arm GNU Toolchain aarch64-none-linux-gnu installed, but don't know how to set it up

When I execute `PS C:\vcpkg> .\vcpkg install qt5:arm64-linux`, I get an error in the location of the detection compiler
Part of the config-arm64-linux-out.log is as follows
[config-arm64-linux-out.log](https://github.com/microsoft/vcpkg/files/8499169/config-arm64-linux-out.log)

```
-- The C compiler identification is unknown
CMake Error at CMakeLists.txt:11 (enable_language):
   No CMAKE_C_COMPILER could be found.

   Tell CMake where to find the compiler by setting either the environment
   variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
   the compiler, or to the compiler name if it is in the PATH.

```
How should I set up the compiler?
The exe path provided after the compiler is installed is as follows
![image](https://user-images.githubusercontent.com/47802547/163659382-2ebaa857-f1e3-4b90-a13c-c6154629ac95.png)
![image](https://user-images.githubusercontent.com/47802547/163659387-213198ea-de16-4252-af51-692bc65a1500.png)

Looking forward to your help

**Comment by WangZP10969** [comment, marked as answer]
I try to use another package gcc-linaro-7.5.0-2019.12-i686-mingw32_aarch64-linux-gnu,and set the environment variable.
I set the path of aarch64-linux-gnu-gcc.exe to the environment variable CC, as well as CMAKE_C_COMPILER, and set the upper directory bin to Path, but there is no difference, vcpkg still does not find the compiler
<img width="120" alt="image" src="https://user-images.githubusercontent.com/47802547/163698923-0be575b7-50cf-428a-9343-299d820da67d.png">

<img width="247" alt="image" src="https://user-images.githubusercontent.com/47802547/163698910-5dc4d7a2-16bc-480b-be4b-3bc6e7dd2e61.png">
What should I do to accomplish this?

**Reply by Osyotr** [reply]
You need to write a proper toolchain file and tell vcpkg to use it.
https://github.com/Microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_chainload_toolchain_file

**Reply by WangZP10969** [reply]
Thanks a lot for the tip, I think I know where to set the path to the compiler
I added the path to the compiler directly in .\vcpkg\scripts\toolchains\linux.cmake
set(CMAKE_CXX_COMPILER "C:/ArmToolChain/bin/aarch64-none-linux-gnu-g++.exe")
set(CMAKE_C_COMPILER "C:/ArmToolChain/bin/aarch64-none-linux-gnu-gcc.exe")
It works, and the library under linux is compiled normally
![image](https://user-images.githubusercontent.com/47802547/163750928-ab89d022-50eb-4957-9cad-94256075e351.png)
I think this is a good start, I will look into other related settings to better control the compilation process.
